src/platform/webos/ConnectivityManagerImpl.cpp

```diff
@@ -935,6 +935,7 @@ ConnectivityManagerImpl::ConnectWiFiNetworkAsync(ByteSpan ssid, ByteSpan credent
 
     VerifyOrReturnError(ssid.size() <= kMaxWiFiSSIDLength, CHIP_ERROR_INVALID_ARGUMENT);
     VerifyOrReturnError(credentials.size() <= kMaxWiFiKeyLength, CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(mWpaSupplicant.iface != nullptr, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Again, why doesn't this require taking the mutex?

Reply:
To be honest I'm not sure about the mutex. There are more functions in there where the `mWpaSupplicant.iface` is used without mutex. I can add mutex locks in all that places, run linux test suite and see whether there will ne no deadlocks.

Reply:
I think we need to figure out what state the mutex is meant to protect and document that, so it's clear when the mutex needs to be locked...

Reply:
Currently I'm trying to incorporate Clang Thread Safety Analysis for this. As I long run I think that we might use this tool in all places where locking is used. Unfortunately, this tool does not show double locks in the same thread if e.g. `funcA` locks mutex and `funcB` locks the same mutex, and in the code we've got `funcA` calling `funcB`. But still, I think it might be very useful for locking analysis.

> I think we need to figure out what state the mutex is meant to protect and document that

OK, I'll try to do that. And speaking of documentation, this clang thread-safety extension might be a great tool for documentation as well - all variables which need locking will have to be correctly annotated with corresponding mutex.